### ISSUE TITLE
Background sync and working offline

### ISSUE THREAD

**Issue by dlukes** [owncloud/ios#1086]
Consider the following workflow:

1. Use Adobe Acrobat to open a PDF file managed by the ownCloud app.
2. Add some annotations to the PDF.
3. Close it.
4. Some time later, reopen the PDF and add some more annotations.

This seems pretty natural and straightforward, but unfortunately, it's currently impossible without inserting an additional step in between 3 and 4, which is opening the ownCloud app in the foreground.

What seems to be happening is that after step 3, ownCloud flags the PDF file for upload, making it impossible to reopen the file until the upload completes, but for some reason, it doesn't actually upload the file in the background. The upload has to be triggered by explicitly opening the ownCloud app, which is cumbersome. Also, less technically sophisticated users might not realize they need to do this, they will just be stuck with a file they can't reopen, and worse, a file which won't be synchronized if they try to open it on a different device.

Additionally, if one is working offline, this workflow is completely impossible, because as soon as the PDF file is flagged for upload, it can't be reopened until the upload completes, but the upload can't complete without going online.

(Technically, you *can* reopen the PDF file if you cancel the upload in the *Uploads* pane of the ownCloud app, but that doesn't just cancel the upload, it also discards the edits you've made and reverts to the previous version of the file, which is a bit surprising; luckily, I didn't lose too much work when I tried this out.)

To summarize this into two concrete suggestions which I think it would greatly improve usability:

1. After opening, editing, saving and closing an ownCloud file in app X, instead of just flagging the file for upload, locking it and requiring the user to open the ownCloud app in order to actually perform the upload, it would be much more convenient if the file was directly uploaded in the background, so that if the user wants to reopen the file in app X at some later point, they can do so directly.
2. It would also be much more convenient if files were locked for opening/editing only when they're actually *in the process* of being uploaded, not when they're just queued for upload. This would make it possible to reopen the same file for multiple editing sessions in app X even when offline.

**Comment by dlukes** [comment]
Thank you for the pointers! I did search the issue tracker before posting but only found issues related to photo/video background sync, so I figured it would make sense to open a new one which specifically emphasizes that the impossibility of opening, closing and reopening an ownCloud file in a different app (without opening the ownCloud app in the meantime) might be confusing for technically unsophisticated users.

> The ownCloud team started a rewrite of the iOS app in February, and an early access release should be available soon…

Neat, I'll be looking forward to that :)
